mos_tests/neutron/python_tests/test_ovs_restart.py

```diff
@@ -0,0 +1,208 @@
+
+            res = self.run_on_vm(vm, vm_keypair, cmd, vm_login="cirros",
+                                 timeout=3 * 60)
+            assert res['exit_code'] == 0, \
+                'Can\'t connect to vm .'.format(vm.name)
+
+        process = Process(target=check_ping_from_vm,
+                          args=(vm, vm_keypair, ip_to_ping))
+
+        try:
+            process.start()
+            yield
+        finally:
+            cmd = 'killall -sigint ping'
+            res = self.run_on_vm(vm, vm_keypair, cmd, vm_login="cirros",
+                                 timeout=3 * 60)
+            assert res['exit_code'] == 0, \
+                'Can\'t kill ping process with pid {}.'.format(res)
+
+            cmd = 'cat /tmp/ovs_restart_ping.log'
+            res = self.run_on_vm(vm, vm_keypair, cmd, vm_login="cirros",
+                                 timeout=3 * 60)
+            assert res['exit_code'] == 0, \
+                'Can\'t get log file /tmp/ovs_restart_ping.log ' \
+                'on vm {}.'.format(vm.name)
+
+            with open('/tmp/ovs_restart_ping.log', 'w') as log_file:
+                for line in res['stdout']:
+                    log_file.write(line)
+
+            process.join(0)
+
+    def test_ovs_restart_pcs_disable_enable(self):
+        """Restart openvswitch-agents with pcs disable/enable on controllers
+
+        Steps:
+            1. Update default security group
+            2. Create router01, create networks net01: net01__subnet,
+            192.168.1.0/24, net02: net02__subnet, 192.168.2.0/24 and
+            attach them to router01.
+            3. Launch vm1 in net01 network and vm2 in net02 network
+            on different computes
+            4. Go to vm1 console and send pings to vm2
+            5. Disable ovs-agents on all controllers, restart service
+            neutron-plugin-openvswitch-agent on all computes, and enable
+            them back. To do this, launch the script against master node
+            6. Check that pings between vm1 and vm2 are not interrupted or
+            not more than 2 packets are lost
+
+        Duration 10m
+
+        """
+        with self.start_ping_from_vm(self.server1, self.instance_keypair,
+                                     self.server2_ip):
+            self.disable_ovs_agents_on_controllers()
+            self.restart_ovs_agents_on_computes()
+
+            # sleep is used to check that system will be stable for some time
+            # after restarting service
+            time.sleep(30)
+
+        with open('/tmp/ovs_restart_ping.log') as f:
+            logs = f.readlines()
+
+        ping_template = r'(\d+) packets transmitted, (\d+) packets received'
+        for line in logs:
+            packets_data = re.search(ping_template, line)
+            if packets_data:
+                sent_count, recieved_count = packets_data.groups()
+                break
+
+        assert packets_data is not None, (
```

Comment:
This will break if no one line in log above.
Will be better to use for...else loop (end call pytest.fail in else)